NewForce-Cohort-8/tabloidfullstack-suplex-syndicate
# Lv deactivate user

### Description
Added ability for admins to deactivate users and prevent those users from logging in
[Trello Ticket 25](https://trello.com/c/DflM7hhd/24-25-deactivate-user-profiles)


### Testing Instructions
1. `git add` and `git commit` whatever you're working on in your branch
2. `git checkout main`
3. `git fetch -a`
4. `git checkout lv-deactivate-user`
5. If you don't see the most recent version of my files `git pull origin lv-deactivate-user`
6. If you have not created the database please do so and follow Steps 7-8 otherwise proceed to Step 9
7. Run this [SQL script](https://github.com/NewForce-Cohort-8/tabloidfullstack-suplex-syndicate/blob/main/SQL/01_Tabloid_Create_DB.sql) to create the initial database
8. Run this [SQL script](https://github.com/NewForce-Cohort-8/tabloidfullstack-suplex-syndicate/blob/main/SQL/02_Tabloid_Seed_Data.sql) to seed the initial data
9. Run these commands in a new SQL query, executing each command one by one
`﻿ALTER TABLE UserProfile ADD UserStatusId int;`
`CREATE TABLE UserStatus (
	Id int,
	Name varChar(50)
);`
`﻿INSERT INTO UserStatus (Id, Name) VALUES (1, 'Active');
INSERT INTO UserStatus(Id, Name) VALUES (2, 'Deactivated');`
`﻿UPDATE UserProfile SET UserStatusId = 1;`
10. Afterwards right click on the UserProfile table in the database and select **View Data** and confirm that a UserStatusId Column has been added and all users have a UserStatusId of 1
11. Right Click on the UserStatus table and select **View Data** to confirm that the data table has been created and seeded with the above information.
12. Open the TabloidFullStack.sln with Visual Studio
13. At the top of Visual Studio, select https as the launcher and click the green play button
14. The Swagger UI should appear in the browser
15. Confirm there is a PUT, a POST, and 3 GET calls under the UserProfile controller
16. If the React app isn't already running, in your terminal `npm start` otherwise refresh the React App
17. Login with the email `foo@bar.com`
18. Select `User Profiles`
19. On Aharon Grzeskowski's row, select the red `Deactivate` button
20. Confirm you have been redirected to **http://localhost:3000/UserProfiles/5/Deactivate**
21. Select the `Cancel` button
22. You should be redirected back to the User Profile list and all users should still have a red Deactivate button in their row
23.  On Aharon Grzeskowski's row, select the red `Deactivate` button
24. Select the red `Deactivate` button
25. You should be redirected back to the User Profile list
26. Confirm that Aharon Grzeskowski no longer has a red Deactivate button in his row
27. Log out
28. try to log in with the email `agrzeskowski3@fc2.comx`
29. You should recieve an invalid email or password warning and still be on the login page

### Checklist
